### PR TITLE
Adding tips and tricks page to MFTF.

### PR DESCRIPTION
**Pull request by dobooth**
## This PR is a:

- New topic

## Summary
Pangolins are internally publishing weekly tips and tricks. This page will be updated with new tips as needed.

When this pull request is merged, it will...
Add a Tips and Tricks page. 
Updates menu structure.

whatsnew
Adds a new [Tips and Tricks](https://devdocs.magento.com/mftf/2.3/tips-tricks.html) page to the MFTF guide.

### CODE REVIEW

File: _data/toc/magento-functional-testing-framework-guide-2_3.yml [hunk 1]
```diff
@@ -95,8 +95,15 @@ pages:
...
 
   - label: Best practices
-    url: /mftf/2.3/best-practices.html
     versionless: true
+    children:
+
+      - label: Best practices
+        url: /mftf/2.3/best-practices.html
+        versionless: true
+      - label: Tips and Tricks
```
Comment (bdenham) [inline _data/toc/magento-functional-testing-framework-guide-2_3.yml]: lowercase tricks to follow sentence case for titles (in styleguide)

File: mftf/2.3/tips-tricks.md [hunk 2]
```diff
@@ -0,0 +1,233 @@
+---
+title: MFTF Tips and Tricks
```
Comment (bdenham) [inline mftf/2.3/tips-tricks.md]: lowercase tricks

File: mftf/2.3/tips-tricks.md [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+    <seeElement selector="{{StorefrontProductMediaSection.productImageFullscreen(ClamberWatch.productImage)}}" stepKey="seeFullscreenProductImage" />
+</test>
+```
+
+## Build selectors in proper order
+
+When building selectors for form elements, start with the parent context of the form element.
+Then specify the element `name` attribute in your selector to ensure the correct element is targeted.
+
+To build selector for an input, use the pattern: `{{section_selector}} {{input_selector}}` or for a button: `{{section_selector}} {{button_selector}}`
```
Comment (bdenham) [inline mftf/2.3/tips-tricks.md]: To build **a** selector for ...

File: mftf/2.3/tips-tricks.md [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+{:style="color:red"}
+BAD:
+
+```xml
+<element name="productName" type="input" selector=".admin__field[data-index=name] input"/>
+```
+
+## Use descriptive stepKey names
+
+Make `stepKeys` descriptive as possible. This helps with readability and clarity.
```
Comment (bdenham) [inline mftf/2.3/tips-tricks.md]: Make `stepKey` **values** as descriptive as possible.

File: mftf/2.3/tips-tricks.md [hunk 5]
```diff
@@ -0,0 +1,233 @@
...
+
+<click selector="{{StorefrontNavigationSection.topCategory(CustomCategory.name)}}" stepKey="clickCategoryLink2" />
+<waitForPageLoad stepKey="waitForPageLoad3" />
+<click selector="{{StorefrontCategoryMainSection.productLinkByHref(CustomSimpleProduct.urlKey)}}" stepKey="clickProductLink2" />
+<waitForPageLoad stepKey="waitForPageLoad4" />
+```
+
+**Exception:**
+
+Use numbers in stepKeys when order is important, such as with testing sort order.
```
Comment (bdenham) [inline mftf/2.3/tips-tricks.md]: Use numbers within `stepKey` values when . . .

Comment (dobooth) [reply mftf/2.3/tips-tricks.md]: Thanks.

File: mftf/2.3/tips-tricks.md (hunk 4 above)
Comment (dobooth) [reply mftf/2.3/tips-tricks.md]: Good.

File: _data/toc/magento-functional-testing-framework-guide-2_3.yml (hunk 1 above)
Comment (dobooth) [reply _data/toc/magento-functional-testing-framework-guide-2_3.yml]: Thanks.

File: mftf/2.3/tips-tricks.md (hunk 2 above)
Comment (dobooth) [reply mftf/2.3/tips-tricks.md]: Fixed.

**Comment by dobooth** [comment]
running tests

**Comment by dobooth** [comment]
running tests

**Comment by dobooth** [comment]
running tests

**Pull request merged**